### ISSUE TITLE
Feature Request: "certbot rename" to change certbot's internal name (not domain or alt domains) for a certificate

### ISSUE THREAD

**Issue by jhgarrison** [certbot/certbot#9736]
Feature Request

Add a `certbot rename` function that adjusts the local "name" for a certificate.  It would

- Rename the archive subdirectory for the cert
- Rename the live subdirectory for the cert
- Adjust the symbolic links from live to archive
- Rename the renewal config file and adjust the directory pointers

See https://community.letsencrypt.org/t/change-certbots-internal-name-for-a-certificate/202119

**Comment by bmw** [comment]
I'm going to flag this as unplanned for now, but I can see how that is useful and I think we should change our mind if we significant demand for this. In the meantime, people can accomplish this by acquiring and installing a new cert with the domain you want and then deleting the old one.

Us offering this is definitely doable, but it's not completely trivial due to our apache and nginx integration. We'd probably need to attempt to install the certificate again under the new name.

**Comment by dj-foxxy** [comment]
This feature would really help amateurs like myself. I've been using Let's Encrypt and certbot since they went public. Back then I didn't know how to create wildcard certs or a single cert for the wildcard and root domain. So I've got lots of  certs with names like `foo.com-000x`  that I'd like to rename to `foo.com` because I have to look up the `-000x` whenever I configuring nginx.

**Comment by dj-foxxy** [comment]
I still think this is a really good idea. Although, I've not needed to it since last year.

**Comment by gflorian** [comment]
I could use a rename function right now: I created a new cert for a service and gave it the service's name. It turns out the service is expecting the cert name to be the FQDN it's serving.

Now I have to revoke the just created cert and request a new one. Doing the steps mentioned in the initial post I am afraid I might miss something and break things.
Doing the revoke and re-request is easy - at least in my case - but feels like kind of an unnecessary effort.

**Comment by revsuine** [comment]
I agree this would be a really useful function. I tried doing this manually a while back and thought all was fine, but it turns out I hadn't updated the symlinks in `/etc/letsencrypt/live` which led to a confusing session of trying to figure out why I was getting errors when trying to renew my cert. Manually renaming things is very human error prone. I could probably write a script to do this more reliably, but most likely it is best handled by a certbot subcommand. And I know I can just request a new cert, but it would be nice to not have to just to not have the numbers at the end of my cert name.

**Comment by ohemorange** [comment]
Unfortunately, attempting to implement a rename function has caused us various problems in the past; see https://github.com/certbot/certbot/issues/10397 for examples.

Instead, we have instructions to rename a certificate using existing functionality now available at https://eff-certbot.readthedocs.io/en/latest/using.html#renaming-certificates
